src/librustc_typeck/collect.rs

```diff
@@ -257,19 +257,9 @@ fn type_param_predicates<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
 
     let item_node_id = tcx.hir.as_local_node_id(item_def_id).unwrap();
     let ast_generics = match tcx.hir.get(item_node_id) {
-        NodeTraitItem(item) => {
```

Comment:
Note the complete replacement of these match statements. All `TraitItem`s and `ImplItem`s contain generics now.

Reply:
Yep. I am happy about how the code is getting more uniform.